atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/encoding/TicketsEncodingStrategy.java

```diff
@@ -57,19 +61,25 @@ public long decodeCellAsStartTimestamp(Cell cell) {
     }
 
     @Override
-    public byte[] encodeCommitTimestampAsValue(long startTimestamp, Long commitTimestamp) {
-        if (commitTimestamp == TransactionConstants.FAILED_COMMIT_TS) {
-            return ABORTED_TRANSACTION_VALUE;
-        }
-        return TransactionConstants.getValueForTimestamp(commitTimestamp - startTimestamp);
+    public byte[] encodeCommitTimestampAsValue(long startTimestamp, TransactionStatus commit) {
+        return TransactionStatuses.caseOf(commit)
+                .committed(ts -> TransactionConstants.getValueForTimestamp(ts - startTimestamp))
+                .aborted(() -> ABORTED_TRANSACTION_VALUE)
+                .otherwise(() -> {
```

Comment:
There are 3 values of aborted txns, empty byte array for persistence, -1 for code and now TransactionStatus.aborted :(